tableaudocumentapi/archivefile.py

```diff
@@ -0,0 +1,67 @@
+import contextlib
+import os
+import shutil
+import tempfile
+import zipfile
+
+import xml.etree.ElementTree as ET
+
+
+@contextlib.contextmanager
+def temporary_directory(*args, **kwargs):
+    d = tempfile.mkdtemp(*args, **kwargs)
+    try:
+        yield d
+    finally:
+        shutil.rmtree(d)
+
+
+def find_file_in_zip(zip, ext):
+    for filename in zip.namelist():
+        if os.path.splitext(filename)[-1].lower() == ext[:-1]:
```

Comment:
Isn't there a way to get the file stream directly from the zipfile without having to extract it?  It might be better to inspect the first tag of the file rather than rely on file extension.  Maybe I'll just open an issue for investigating that as an option.